code/modules/uplink/uplink_item.dm

```diff
@@ -184,6 +184,14 @@ var/list/uplink_items = list() // Global list so we only initialize this once.
 	surplus = 25
 	include_modes = list(/datum/game_mode/nuclear)
 
+/datum/uplink_item/dangerous/mgl
+	name = "6B40 MGL"
+	desc = "New Russian semi-automatic grenade launcher. Has 6 chambers 40mm grenades."
+	item = /obj/item/weapon/gun/projectile/shotgun/automatic/mgl
+	cost = 25
```

Comment:
You're right. I think this might be the only gun that's going to break the 30 TC cost next round of commits.